### PR TITLE
feat(security-center): Add Resource v2 API Assets Security Marks Samples

### PR DESCRIPTION
**Pull request by vijaykanthm**
## Description

This PR adds v2 API Assets Security Marks Node js client samples to Add Security Marks, Delete Security Marks, Add Delete Security Marks.

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed guidelines from [CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md) and [Samples Style Guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [ ] **Tests** pass:   `npm test` (see [Testing](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#run-the-tests-for-a-single-sample))
- [ ] **Lint** pass:   `npm run lint` (see [Style](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#style))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This pull request is from a branch created directly off of `GoogleCloudPlatform/nodejs-docs-samples`. Not a fork.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new sample directory, and I created [GitHub Actions workflow](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/CONTRIBUTING.md#adding-new-samples) for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: security-center/snippets/system-test/v2/assetSecurityMarks.test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  let data;
+  before(async () => {
+    // Creates a new client.
+    const client = new SecurityCenterClient();
+
+    const [assetResults] = await client.listAssets({
+      parent: client.organizationPath(organizationId),
+    });
+    const randomAsset =
+      assetResults[Math.floor(Math.random() * assetResults.length)].asset;
```
Comment (feywind) [inline security-center/snippets/system-test/v2/assetSecurityMarks.test.js:36]: Seeing `random()` in a test makes me worry that it's going to end up being flaky, but I guess if all the items are more or less equivalent, should be okay.

File: security-center/snippets/system-test/v2/assetSecurityMarks.test.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    assert.match(output, /key_a/);
+    assert.match(output, /value_a/);
+    assert.match(output, /key_b/);
+    assert.match(output, /value_b/);
+    assert.notMatch(output, /undefined/);
+  });
+
+  it('client can add and delete security marks', () => {
+    // Ensure marks are set.
+    exec(`node v2/addSecurityMarks.js ${data.assetName}`);
```
Comment (feywind) [inline security-center/snippets/system-test/v2/assetSecurityMarks.test.js:56]: Should these be execSync()?

**Review by feywind — COMMENTED** [review]
This generally looks fine, maybe just check on that exec() vs execSync() thing. Let's see if we can get someone closer to docs/samples to approve the rest of the way.

File: security-center/snippets/system-test/v2/assetSecurityMarks.test.js (hunk 2 above)
Comment (vijaykanthm) [reply security-center/snippets/system-test/v2/assetSecurityMarks.test.js]: Addressed. Other tests were using the exec() method so I have used the same but I have updated it.

File: security-center/snippets/system-test/v2/assetSecurityMarks.test.js (hunk 1 above)
Comment (vijaykanthm) [reply security-center/snippets/system-test/v2/assetSecurityMarks.test.js]: Refactored it.

**Comment by iennae** [comment]
@gemini review

File: security-center/snippets/system-test/v1/notifications.test.js [hunk 3]
```diff
@@ -36,37 +36,51 @@ describe('Client with Notifications', async () => {
...
+          configId: configId,
+          notificationConfig: {
+            description: 'Sample config for node.js',
+            pubsubTopic: pubsubTopic,
+            streamingConfig: {filter: 'state = "ACTIVE"'},
+          },
+        });
+      } catch (err) {
+        console.error(`Error creating config ${configId}:`, err.message);
+      }
```
[the remark this answers is not in the document]
Comment (vijaykanthm) [reply security-center/snippets/system-test/v1/notifications.test.js]: Addressed

File: security-center/snippets/system-test/v1/notifications.test.js [hunk 4]
```diff
@@ -36,37 +36,51 @@ describe('Client with Notifications', async () => {
...
     }
 
     await createNotificationConfig(deleteConfig);
+    await new Promise(resolve => setTimeout(resolve, 1000)); // 1-second delay
     await createNotificationConfig(getConfig);
+    await new Promise(resolve => setTimeout(resolve, 1000)); // 1-second delay
     await createNotificationConfig(listConfig);
+    await new Promise(resolve => setTimeout(resolve, 1000)); // 1-second delay
     await createNotificationConfig(updateConfig);
```
[the remark this answers is not in the document]
Comment (vijaykanthm) [reply security-center/snippets/system-test/v1/notifications.test.js]: Addressed

File: security-center/snippets/system-test/v1/notifications.test.js [hunk 5]
```diff
@@ -36,37 +36,51 @@ describe('Client with Notifications', async () => {
...
-      await client.deleteNotificationConfig({name: name});
+      try {
+        const name = client.organizationNotificationConfigPath(
+          organizationId,
+          configId
+        );
+        await client.deleteNotificationConfig({name: name});
+      } catch (err) {
+        console.error(`Error deleting config ${configId}:`, err.message);
+      }
```
[the remark this answers is not in the document]
Comment (vijaykanthm) [reply security-center/snippets/system-test/v1/notifications.test.js]: Addressed

File: security-center/snippets/system-test/v1/notifications.test.js [hunk 6]
```diff
@@ -36,37 +36,51 @@ describe('Client with Notifications', async () => {
...
     }
 
     await deleteNotificationConfig(createConfig);
+    await new Promise(resolve => setTimeout(resolve, 1000)); // 1-second delay
     await deleteNotificationConfig(getConfig);
+    await new Promise(resolve => setTimeout(resolve, 1000)); // 1-second delay
     await deleteNotificationConfig(listConfig);
+    await new Promise(resolve => setTimeout(resolve, 1000)); // 1-second delay
     await deleteNotificationConfig(updateConfig);
```
[the remark this answers is not in the document]
Comment (vijaykanthm) [reply security-center/snippets/system-test/v1/notifications.test.js]: Addressed

File: security-center/snippets/v2/addSecurityMarks.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    const [newMarks] = await client.updateSecurityMarks({
+      securityMarks: {
+        name: `${assetName}/securityMarks`,
+        marks: {key_a: 'value_a', key_b: 'value_b'},
+      },
+      // Only update the marks with these keys.
+      updateMask: {paths: ['marks.key_a', 'marks.key_b']},
+    });
+
+    console.log('New marks: %', newMarks);
```
[the remark this answers is not in the document]
Comment (vijaykanthm) [reply security-center/snippets/v2/addSecurityMarks.js]: Addressed

File: security-center/snippets/v2/addSecurityMarks.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  async function addSecurityMarks() {
+    // assetName is the full resource path for the asset to update.
+    /*
+     * TODO(developer): Uncomment the following lines
+     */
+    // Specify the value of 'assetName' in one of the following formats:
+    //    `organizations/${org-id}/assets/${asset-id}`;
+    //    `projects/${project-id}/assets/${asset-id}`;
+    //    `folders/${folder-id}/assets/${asset-id}`;
+    // const assetName = "organizations/123123342/assets/12312321";
```
[the remark this answers is not in the document]
Comment (vijaykanthm) [reply security-center/snippets/v2/addSecurityMarks.js]: Addressed

**Pull request merged**